spring-integration-core/src/main/java/org/springframework/integration/config/xml/AggregatorParser.java

```diff
@@ -60,24 +58,24 @@ protected BeanDefinitionBuilder parseHandler(Element element, ParserContext pars
 			else {
 				processor = new RuntimeBeanReference(ref);
 			}
-			builder.addConstructorArgValue(processor);
+			builder.addPropertyValue("processorBean", processor);
 		}
 		else {
 			if (StringUtils.hasText(element.getAttribute(EXPRESSION_ATTRIBUTE))) {
 				String expression = element.getAttribute(EXPRESSION_ATTRIBUTE);
 				BeanDefinitionBuilder adapterBuilder = BeanDefinitionBuilder.genericBeanDefinition(ExpressionEvaluatingMessageGroupProcessor.class);
```

Comment:
If we polish here already, let's make this piece of code readable - line length 120.
Plus `@Copyright` update.